discord_bot/discord_bot.py

```diff
@@ -34,53 +36,94 @@ def load_config():
         with open(CONFIG_FILE_PATH) as f:
             return json.load(f)
 
-    def get_gateway(self):
+    def get_gateway(self, **kwargs):
         """
-        Caches a gateway value and retrieves a new URL
+        Caches a gateway value, authenticates, and retrieves a new URL
         :return: gateway URL
         """
-        r = requests.get(self.config["discord_gateway_endpoint"])
+        header = {
+            "headers": {
+                "Authorization": "Bot {}".format(self.config["handshake_identity"]["token"]),
+                "User-Agent": "DiscordBot (https://github.com/ShifuYee/Discord-Bot, 0.0.1)"
+            }
+        }
+        kwargs = dict(header, **kwargs)
+        r = requests.get(self.config["discord_gateway_endpoint"], **kwargs)
+        assert 200 == r.status_code, r.reason
         return r.json()['url']
 
     async def send_json(self, payload):
         asyncio.ensure_future(self.websocket.send(json.dumps(payload)))
 
     async def handshake(self, message):
         """
-        When a websocket connection is opened, Hello payload is received. Then, an Identify payload is sent as
+        When a websocket connection is opened, Hello payload is received. Then, an Identify/Resume payload is sent as
         part of the handshake to authorize this client.
 
         :param message: dict - hello payload
         """
-        self.heartbeat_interval_ms = message["d"]["heartbeat_interval"]
+        self.heartbeat_interval_ms = message["d"]["heartbeat_interval"]  # seconds
         asyncio.ensure_future(self.heartbeat())
-
         handshake_identity = self.config["handshake_identity"]
-        asyncio.ensure_future(self.send_json({"op": Opcodes.IDENTIFY, "d": handshake_identity}))
+        if not self.session_id:
```

Comment:
Personally I find it more understandable the less `not` values that are used, so I would switch the order to use `if self.session_id` first.